src/EoS/Nuclear/NuclearEoS.cpp

```diff
@@ -253,15 +252,17 @@ void nuc_eos_C_short( real *Out, const real *In,
 #     if ( NUC_EOS_SOLVER != NUC_EOS_SOLVER_ORIG )
       const real *table_Aux = alltables_Aux + var_idx*nrho_Aux*nmode_Aux*nye_Aux;
 
-#     if   ( NUC_EOS_SOLVER == NUC_EOS_SOLVER_LUT    )
-      findtoreps( lr, var_mode, xye, &ltoreps, table_Aux, nrho_Aux, nmode_Aux, nye_Aux, ntoreps,
-                  logrho_Aux, mode_Aux, yes_Aux, logtoreps, IntScheme_Aux, keyerr );
-
-#     elif ( NUC_EOS_SOLVER == NUC_EOS_SOLVER_DIRECT )
+      if ( *keyerr == 0 )
```

Comment:
If we use `keyerr`, the condition here should be `*keyerr != 0`. However, we also need to reset the `keyerr` to `0` somewhere, otherwise the NR + Bisect method will always be invoked.